app/models/resource_action_workflow.rb

```diff
@@ -32,18 +32,19 @@ def add_request_to_cart
 
   def process_request(state)
     validate_dialog.tap do |result|
-      generate_request(state) if result[:errors].blank?
```

Comment:
@imtayadeway Not a big fan of the nested if-statements this created.  I took at another try at this and broke the methods up a little differently.  Let me know what you think.

```ruby
  def process_request(state)
    result = {:errors => validate_dialog}
    return result unless result[:errors].blank?

    values = create_values
    if create_request?(values)
      result[:request] = generate_request(state, values)
    else
      ra = load_resource_action(values)
      ra.deliver_to_automate_from_dialog(values, @target, @requester)
    end

    result
  end

  def generate_request(state, values)
    create_request(values).tap do |request|
      process_service_order(request, state) if request
    end
  end
...

  def validate_dialog
    @dialog.validate_field_data
  end
```
Note: The validate_dialog method, which is not called anywhere else, was changed to just return the errors and not generate the hash wrapper.